### PR TITLE
Dedicated instance for /oai requests

### PR DESCRIPTION
**Pull request by philpep**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -68,6 +70,71 @@ Requirements
...
 
+Recomended usage
+================
+
+Given target minion_id is 'srv'.
+
+/srv/salt/top.sls::
+
+    base:
+        'srv'
```
Comment (dlax) [inline README.rst]: trailing `:` missing

File: README.rst [hunk 2]
```diff
@@ -68,6 +70,71 @@ Requirements
...
+/srv/pillar/saemref.sls::
+
+    saemref:
+        lookup:
+            instance:
+                base_url: http://saemref.example.com
+                # generated with: pwgen -sy 32
+                sessions_secret: "iajI*=^m;oqNc7:%ta'A6r3BDLUjB^,y"
+                authtk_session_secret: ")E6"LT,XMUd."Ekg+(<2m%RVqLxdV[a4"
+                authtk_persistent_secret: "Ay]%v4w>B+yP$|~BLTL^+785(6@=,Gk$"
```
Comment (dlax) [inline README.rst]: I'm not sure we should recommend non-alphanumeric characters. I had trouble with these in paste's INI parser recently. (/me anticipating migration)

File: README.rst [hunk 3]
```diff
@@ -68,6 +70,71 @@ Requirements
...
+
+/srv/pillar/top.sls::
+
+    base:
+        'srv':
+            - saemref
+
+
+/srv/pillar/saemref.sls::
+
```
Comment (dlax) [inline README.rst]: What's the value of repeating this here, over pointing to `pillar.example`? It looks like repetition that would quickly get out of sync...

File: README.rst [hunk 4]
```diff
@@ -68,6 +70,71 @@ Requirements
...
+                pass: uOCWIzj4
+            admin:
+                login: admin
+                pass: qj0FOtM6UC
+
+    postgres:
+        version: 9.4
+
+
+Example of reverse proxy configuration for nginx::
```
Comment (dlax) [inline README.rst]: I'm pretty sure our clients don't use nginx (unfortunately).

File: README.rst [hunk 5]
```diff
@@ -68,6 +70,71 @@ Requirements
...
+        location /oai {
+            proxy_pass http://srv:8081;
+        }
+    }
+
+
+For the first installation run ``salt srv state.sls db-create`` to create
+the database (WARNING: this will destroy existing database if exists).
+
+Then run: ``salt srv state.highstate`` to finish the installation.
```
Comment (dlax) [inline README.rst]: Shouldn't it be the other way around? (highstate before)

Comment (philpep) [inline README.rst]: `highstate` never run `db-create`, so running highstate before would try to start the instance without database.

File: README.rst (hunk 4 above)
Comment (philpep) [inline README.rst]: Is it ok to keep this example with nginx ? I'd be nice to add an apache config too, unfortunately I don't know the syntax.

File: README.rst (hunk 3 above)
Comment (philpep) [inline README.rst]: Ok replaced by a link to `pillar.example`.

**Comment by dlax** [comment]
Philippe Pepiot a écrit :
> *@philpep* commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In README.rst
> <https://github.com/logilab/saemref-formula/pull/48#discussion_r104697026>:
>
>> +    server {
> +        listen 80;
> +        server_name saemref.example.com;
> +        location / {
> +            proxy_pass http://srv:8080;
> +        }
> +        location /oai {
> +            proxy_pass http://srv:8081;
> +        }
> +    }
> +
> +
> +For the first installation run ``salt srv state.sls db-create`` to create
> +the database (WARNING: this will destroy existing database if exists).
> +
> +Then run: ``salt srv state.highstate`` to finish the installation.
>
> |highstate| never run |db-create|, so running highstate before would try
> to start the instance without database.

Then the text is misleading and should indicate that running db-create
states would not just create the database, but also install the
application when run the first time.

**Comment by dlax** [comment]
Philippe Pepiot a écrit :
> *@philpep* commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In README.rst
> <https://github.com/logilab/saemref-formula/pull/48#discussion_r104697955>:
>
>> +            db:
> +                driver: postgres
> +                host: pg.example.com
> +                port: 5432
> +                name: saemref
> +                user: saemref
> +                pass: uOCWIzj4
> +            admin:
> +                login: admin
> +                pass: qj0FOtM6UC
> +
> +    postgres:
> +        version: 9.4
> +
> +
> +Example of reverse proxy configuration for nginx::
>
> Is it ok to keep this example with nginx ? I'd be nice to add an apache
> config too, unfortunately I don't know the syntax.
>

Doesn't hurt, so yes.

**Comment by dlax** [comment]
Build failure seems unrelated but this does not mean these changes are working.
I triggered a new build on master (https://travis-ci.org/logilab/saemref-formula/builds/206659252)

**Pull request merged**